module/executiondatasync/optimistic_sync/execution_result/info_provider.go

```diff
@@ -112,6 +134,34 @@ func (e *Provider) ExecutionResultInfo(
 	}, nil
 }
 
+// validateRequiredExecutors verifies that the provided set of execution node
+// identities contains all nodes required for processing and that the requested
+// number of required executors does not exceed the available number.
+//
+// Expected errors during normal operations:
+//   - [common.RequiredExecutorsCountExceededError]: Required executor IDs count exceeds available executors.
+//   - [common.UnknownRequiredExecutorError]: A required executor ID is not in the available set.
+func (e *Provider) validateRequiredExecutors(
+	required flow.IdentifierList,
+	available flow.IdentityList,
+) error {
+	if len(available) < len(required) {
```

Comment:
I think this should include a check that `AgreeingExecutorsCount` is less than the available executors